builtin/providers/aws/resource_aws_db_event_subscription.go

```diff
@@ -0,0 +1,326 @@
+			SubscriptionName: aws.String(d.Id()),
+			Enabled:          aws.Bool(d.Get("enabled").(bool)),
+			SnsTopicArn:      aws.String(d.Get("sns_topic").(string)),
+			SourceType:       aws.String(d.Get("source_type").(string)),
+			EventCategories:  eventCategories,
+		})
+
+		if err != nil {
+			return fmt.Errorf("Modifying RDS Event Subscription %s failed: %s", d.Id(), err)
+		}
+
+		log.Println(
+			"[INFO] Waiting for RDS Event Subscription modification to finish")
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"modifying"},
+			Target:     []string{"active"},
+			Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta.(*AWSClient).rdsconn),
+			Timeout:    40 * time.Minute,
+			MinTimeout: 10 * time.Second,
+			Delay:      30 * time.Second, // Wait 30 secs before starting
+		}
+
+		// Wait, catching any errors
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf("Modifying RDS Event Subscription %s failed: %s", d.Id(), err)
+		}
+	}
+
+	if arn, err := buildRDSEventSubscriptionARN(d, meta); err == nil {
+		if err := setTagsRDS(rdsconn, d, arn); err != nil {
+			return err
+		} else {
+			d.SetPartial("tags")
+		}
+	}
+	d.Partial(false)
+
+	return nil
+}
+
+func resourceAwsDbEventSubscriptionDelete(d *schema.ResourceData, meta interface{}) error {
+	rdsconn := meta.(*AWSClient).rdsconn
+	deleteOpts := rds.DeleteEventSubscriptionInput{
+		SubscriptionName: aws.String(d.Id()),
+	}
+
+	if _, err := rdsconn.DeleteEventSubscription(&deleteOpts); err != nil {
+		rdserr, ok := err.(awserr.Error)
+		if !ok {
+			return fmt.Errorf("Error deleting RDS Event Subscription %s: %s", d.Id(), err)
+		}
+
+		if rdserr.Code() != "DBEventSubscriptionNotFoundFault" {
+			return fmt.Errorf("Error deleting RDS Event Subscription %s: %s", d.Id(), err)
+		}
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"deleting"},
+		Target:     []string{},
+		Refresh:    resourceAwsDbEventSubscriptionRefreshFunc(d, meta.(*AWSClient).rdsconn),
+		Timeout:    40 * time.Minute,
+		MinTimeout: 10 * time.Second,
+		Delay:      30 * time.Second, // Wait 30 secs before starting
+	}
+	_, err := stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting RDS Event Subscription %s: %s", d.Id(), err)
+	}
+	return err
+}
+
+func resourceAwsDbEventSubscriptionRefreshFunc(
+	d *schema.ResourceData,
+	rdsconn *rds.RDS) resource.StateRefreshFunc {
+
+	return func() (interface{}, string, error) {
+		sub, err := resourceAwsDbEventSubscriptionRetrieve(d, rdsconn)
+
+		if err != nil {
+			log.Printf("Error on retrieving DB Event Subscription when waiting: %s", err)
+			return nil, "", err
+		}
+
+		if sub == nil {
+			return nil, "", nil
+		}
+
+		if sub.Status != nil {
+			log.Printf("[DEBUG] DB Event Subscription status for %s: %s", d.Id(), *sub.Status)
+		}
+
+		return sub, *sub.Status, nil
+	}
+}
+
+func buildRDSEventSubscriptionARN(d *schema.ResourceData, meta interface{}) (string, error) {
```

Comment:
Similar to other functions I called out earlier, would you mind reducing the interface to the only necessary arguments? i.e. 

```go
func buildRDSEventSubscriptionARN(customerAwsId, subscriptionId string) (string) {
```

Reducing these interfaces is not just nitpicking. It makes such functions a lot easier to test (test-ability is important even if we aren't currently testing this function anywhere). It also makes it easier for readers to understand how such functions operate - i.e. if a function accepts 2 strings and returns a string, it is quite obvious that it's going to be some kind of concatenation not involving any API calls or other heavy operations and I can _fairly safely_ assume this without having to look at the actual implementation of it.